### PR TITLE
Retry image stream updates when pruning images

### PR DESCRIPTION
**Pull request by soltysh**
@mfojtik @brenton ptal, this is a bug fix so I guess we can merge it at any point

@miminar fyi

### CODE REVIEW

**Comment by soltysh** [comment]
@fabianofranz need your sign off as well

**Comment by mfojtik** [comment]
@soltysh the workaround before this fix is basically just re-run the pruning? i'm fine retrying on update conflicts.

/approve no-issue

**Comment by fabianofranz** [comment]
> this is a bug fix so I guess we can merge it at any point

what's the bug or issue?

**Comment by soltysh** [comment]
> what's the bug or issue?

This is part of https://bugzilla.redhat.com/show_bug.cgi?id=1408676 when this error happens:
```
error updating image stream xzy/qwerty to remove references to image sha256:12345: imagestreams "xyz" cannot be updated: the object has been modified; please apply your changes to the latest version and try again
```

**Comment by soltysh** [comment]
Flake #15874
/retest

**Comment by fabianofranz** [comment]
@soltysh tks!
/lgtm

File: pkg/oc/admin/prune/images.go [hunk 1]
```diff
@@ -58,9 +58,9 @@ var (
 		Insecure connection is allowed in the following cases unless certificate-authority is
 		specified:
 
-		 1. --force-insecure is given  
-		 2. provided registry-url is prefixed with http://  
-		 3. registry url is a private or link-local address  
+		 1. --force-insecure is given
```
Comment (miminar) [inline pkg/oc/admin/prune/images.go]: This change breaks formatting. The spaces were there to preserve line breaks. Without them, the lines will be concatenated.

**Pull request merged**